cmd/posix.go

```diff
@@ -367,30 +364,36 @@ func (s *posix) getVolDir(volume string) (string, error) {
 	return volumeDir, nil
 }
 
-// checkDiskFound - validates if disk is available,
-// returns errDiskNotFound if not found.
-func (s *posix) checkDiskFound() (err error) {
-	if !s.IsOnline() {
-		return errDiskNotFound
+func (s *posix) getDiskID() (string, error) {
+	s.RLock()
+	diskID := s.diskID
+	s.RUnlock()
+
+	formatFile := pathJoin(s.diskPath, minioMetaBucket, formatConfigFile)
+	fi, err := os.Stat(formatFile)
+	if err != nil {
+		// If the disk is still not initialized.
+		return "", err
+	}
+
+	if os.SameFile(fi, s.formatFileInfo) && fi.ModTime() == s.formatFileInfo.ModTime() {
```

Comment:
SameFile validates the modtime as well @krishnasrinivas

Reply:
also shouldn't this be inside read lock?

Reply:
```
29func sameFile(fs1, fs2 *fileStat) bool {                                                                                                                                                                         
30        return fs1.sys.Dev == fs2.sys.Dev && fs1.sys.Ino == fs2.sys.Ino                                                                                                                                          
31}                                                                                                                                                                                                                
```

doesnt check the modTime

> also shouldn't this be inside read lock?
Currently lock() call is just protecting access to `s.diskID`
It is OK, not to do it inside lock because 100% correct way to do it would be to:
```
fd = open(".minio.sys/format.json")
fcntl(fd, FD_LK)
verifyDiskID()
doStorageAPICall()
fcntl(fd, FD_UNLK)
close(fd)
```

So that the storage-api call happens on the right disk. But I think we can just avoid this overhead and we can just do a crude best effort approach.

Reply:
oh dam! perhaps you can write a new call called `sameFile` which checks both os.FileInfo properly including the Size and ModTime, permissions as well? @krishnasrinivas